src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -4469,6 +4496,19 @@ theorem toNat_smod {x y : BitVec w} : (x.smod y).toNat =
   <;> simp only [umod, toNat_eq, toNat_ofNatLT, toNat_ofNat, Nat.zero_mod] at h'' h'''
   <;> simp
 
+theorem toFin_smod {x y : BitVec w} : (x.smod y).toFin =
+    match x.msb, y.msb with
+    | false, false => x.toFin % y.toFin
+    | false, true => if x % -y = 0#w then 0 else x.toFin % (-y).toFin + y.toFin
+    | true, false => if -x % y = 0#w then 0 else y.toFin - (-x).toFin % y.toFin
+    | true, true => (-(-x % -y)).toFin := by
+  simp only [smod_eq]
```

Comment:
```lean
theorem toFin_smod {x y : BitVec w} : (x.smod y).toFin =
    ...
    | false, true => if x % -y = 0#w then 0 else x.toFin % (-y).toFin + y.toFin
    | true, false => if -x % y = 0#w then 0 else y.toFin - (-x).toFin % y.toFin
```

Is there a nicer characterization available for these two cases? I have not thought about this, but I would hope that there is a "uniform" statement, (a) without a case split, and (b) without the mixture of `%` on `bitvec % bitvec` (from `x % -y`) and `Fin % Fin` (from `x.toFin % (-y).toFin`)